File: src/com/zypex/piflow/path/PathFollower.java
package com.zypex.piflow.path;

import com.zypex.piflow.DrivePowers;
import com.zypex.piflow.DrivetrainConfig;
import com.zypex.piflow.path.Path;
import com.zypex.piflow.profile.Profile;
import com.zypex.piflow.profile.ProfileBuilder;

public class PathFollower {

    private final Profile profile;
    private final Path path;
    private final DrivetrainConfig config;
    public int pathSection;

    public PathFollower(DrivetrainConfig config, Path path){
        this.config = config;
        this.path = path;
        profile = new ProfileBuilder(config, path).build();
    }

    public DrivePowers getPowers(double x, double y){
        return null;
    }

}

File: src/com/zypex/piflow/path/PathFollower.kt
package com.zypex.piflow.path

import com.zypex.piflow.DrivePowers
import com.zypex.piflow.DrivetrainConfig
import com.zypex.piflow.profile.Profile
import com.zypex.piflow.profile.ProfileBuilder

class PathFollower(private val config: DrivetrainConfig, private val path: Path) {
    private val profile: Profile?
    var pathSection = 0

    init {
        profile = ProfileBuilder(config, path).build()
    }

    fun getPowers(x: Double, y: Double): DrivePowers? {
        return null
    }
}